src/pw/fpga/opencl_utils.c

```diff
@@ -0,0 +1,370 @@
+  cl_platform_id *pids = (cl_platform_id*) malloc(sizeof(cl_platform_id) * num_platforms);
+  status = clGetPlatformIDs(num_platforms, pids, NULL);
+  if (status != CL_SUCCESS){
+    printf("Query for platform ids failed\n");
+    exit(0);
+  }
+
+  // Convert argument string to lowercase compare platform names
+  size_t pl_len = strlen(platform_name);
+  char name_search[pl_len + 1];
+  tolowercase(platform_name, name_search);
+
+  // Search the platforms for the platform name passed as argument
+  size_t sz;
+  for(i=0; i<num_platforms; i++){
+    // Get the size of the platform name referred to by the id
+		status = clGetPlatformInfo(pids[i], CL_PLATFORM_NAME, 0, NULL, &sz);
+    if (status != CL_SUCCESS){
+      printf("Query for platform info failed\n");
+      free(pids);
+      exit(0);
+    }
+
+    char pl_name[sz];
+    char plat_name[sz];
+
+    // Store the name of string size
+	  status = clGetPlatformInfo(pids[i], CL_PLATFORM_NAME, sz, pl_name, NULL);
+    if (status != CL_SUCCESS){
+      printf("Query for platform info failed\n");
+      free(pids);
+      exit(0);
+    }
+
+    tolowercase(pl_name, plat_name);
+    if( strstr(plat_name, name_search)){
+      cl_platform_id pid = pids[i];
+      free(pids);
+      return pid;
+    }
+  }
+  free(pids);
+  return NULL;
+}
+
+/******************************************************************************
+ * \brief   returns the list of all devices for the specfic platform
+ * \param   platform id to search for devices 
+ * \param   specific type of device to search for
+ * \param   total number of devices found for the given platform
+ * \retval  array of device ids
+ *****************************************************************************/
+cl_device_id* getTestDevices(cl_platform_id pid, cl_device_type device_type, cl_uint *num_devices) {
+  cl_int status;
+
+  // Query for number of devices
+  status = clGetDeviceIDs(pid, device_type, 0, NULL, num_devices);
+  if(status != CL_SUCCESS){
+    printf("Query for number of devices failed\n");
+    exit(0);
+  }
+
+  //  Based on the number of devices get their device ids
+  cl_device_id *dev_ids = (cl_device_id*) malloc(sizeof(cl_device_id) * (*num_devices));
+  status = clGetDeviceIDs(pid, device_type, *num_devices, dev_ids, NULL);
+  if(status != CL_SUCCESS){
+    printf("Query for device ids failed\n");
+    free(dev_ids);
+    exit(0);
+  }
+  return dev_ids;
+}
+
+/******************************************************************************
+ * \brief   returns the list of all devices for the specfic platform
+ * \param   context created using device
+ * \param   array of devices
+ * \param   number of devices found
+ * \param   size of FFT3d
+ * \retval  created program
+ *****************************************************************************/
+cl_program getProgramWithBinary(cl_context context, const cl_device_id *devices, unsigned num_device, int N[3], char *data_path){
+  char bin_path[500];
```

Comment:
harcoded buffer size, please use a `const int` at the beginning with a meaningful name